### DISCUSSION TITLE
GCM expansion

### DISCUSSION THREAD

**Discussion by hcorson-dosch-usgs** [DOI-USGS/lake-temperature-process-models#70] [General]
@lindsayplatt  - Summary of GCM model runs for the expanded footprint, excluding SD and ND lakes

_________________________

I just ran the GCM models for the full footprint (12,864 lakes). For now, I'm only summarizing results for lakes NOT in SD or ND (12,103 lakes), b/c we've not yet excluded lakes in those states that fall outside the GCM footprint.

Across all lakes outside of the Dakotas, all 18 model runs succeeded for 11,671 lakes (96.4%). 1+ model runs failed for 432 lakes (3.6% - 1923 failed runs in all).

The mean lake depth across the 432 lakes outside of the Dakotas for which GCM runs failed is 1.6 meters. Median lake depth is 1 m, minimum lake depth is 0 m (4 sites - `c("nhdhr_113236931", "nhdhr_151870592", "nhdhr_151874009", "nhdhr_45361479" )`), maximum lake depth is 53.6 m (`nhdhr_129226206`). There were 6 lakes over 10 meters deep that failed. 

The 1923 run failures for 432 lakes were primarily due to NA values and model layers dropping to 1:
```r
> p2_gcm_glm_uncalibrated_runs  %>% left_join(p1_lake_cell_tile_xwalk_df, by='site_id') %>% filter(!(state %in% c('SD', 'ND')), glm_success==FALSE) %>% group_by(glm_code, extraction_error) %>% summarize(n = n())
`summarise()` has grouped output by 'glm_code'. You can override using the `.groups` argument.
# A tibble: 6 × 3
# Groups:   glm_code [4]
  glm_code extraction_error                                                    n
     <int> <chr>                                                           <int>
1        0 NA temperature values                                             953
2        0 Number of layers drops to 1                                       933
3        0 Number of layers drops to 1, Maximum hice value exceeds lake d…     1
4        1 NA                                                                 26
5      134 NA                                                                  8
6      139 NA                                                                  2
```

Note that the 'NA temperature values' error is triggered by [the extraction of GLM temperatures](https://github.com/USGS-R/lake-temperature-process-models/blob/main/2_run/src/run_glm3.R#L17) and precedes [the generation of the other two extraction errors](https://github.com/USGS-R/lake-temperature-process-models/blob/main/2_run/src/run_glm3.R#L33-L48). So lakes for which temperature data extraction failed (`NA temperature values`) may have also have excessive ice generation or only 1 simulation layer.

**Comment by hcorson-dosch-usgs** [comment]
### Comparison to latest NLDAS model runs

In the [latest NLDAS run for the full footprint](https://github.com/USGS-R/lake-temperature-process-models/discussions/57#discussioncomment-2867189), model runs for 11990/12103 lakes outside of the Dakotas succeeded (99%). Model runs failed for 113/12103 lakes outside of the Dakotas (1%).  

There were 12 lakes for which the NLDAS run failed but the GCM runs did not.
There were 331 lakes for which the GCM runs failed but the NLDAS runs did not.
There were 101 lakes for which both NLDAS and GCM runs failed.

That means there were either NLDAS or GCM failures across 444 lakes. 

**So currently we have NLDAS and all GCM simulation results for 11,659 lakes outside of the Dakotas, or 96%**

**Comment by hcorson-dosch-usgs** [comment]
@lindsayplatt @jread-usgs 

The GCM evaluation plots are [HERE](https://doimspp.sharepoint.com/:f:/r/sites/IIDDStaff/Shared%20Documents/Project%20-%20Modeling%20Lakes/GLM/lake-temperature-process-models/GCM_evaluation_plots/FullFootprint_20220610?csf=1&web=1&e=ezB0ht). There were 1,175 evaluation sites (sites with >= 10 min_obs_dates, successful model runs, and matched pred-obs) in the GCM contemporary period (1981-2000).

**Reply by lindsayplatt** [reply]
I am surprised by those 30 deg temp bins for the bottom depths. I wonder if those are for shallow lakes.

**Reply by hcorson-dosch-usgs** [reply]
Not sure re: depth w/o some digging, but there are definitely fewer dates and sites in those more extreme temp bins, _especially_ at depth:
```r
> colnames(p5_gcm_bias_temp)
[1] "driver"      "depth_class" "temp_bin"    "bias"        "n_dates"
[6] "n_sites"
> p5_gcm_bias_temp %>% group_by(temp_bin, depth_class) %>% summarize(across(n_dates:n_sites, .fns=list(mean = mean)))
`summarise()` has grouped output by 'temp_bin'. You can override using the `.groups` argument.
# A tibble: 34 × 4
# Groups:   temp_bin [10]
   temp_bin depth_class n_dates_mean n_sites_mean
      <dbl> <fct>              <dbl>        <dbl>
 1        0 surface             284          99
 2        0 bottom                6           5
 3        2 surface            2277.        265
 4        2 bottom              356.         51.5
 5        4 surface            1317.        264
 6        4 bottom             2258.        180.
 7        6 surface            1206.        305
 8        6 bottom             5632.        327.
 9        8 surface            1784.        455
10        8 bottom             5948         424.
11       10 surface            2348.        581
12       10 bottom             4790.        436.
13       12 surface            2752         617
14       12 bottom             2979.        395.
15       14 surface            3407.        746.
16       14 bottom             2368.        361
17       16 surface            4320         827
18       16 bottom             1776.        357.
19       18 surface            5468.        922
20       18 bottom             1432.        301.
21       20 surface          7187         1037
22       20 bottom           1225          286.
23       22 surface         10148.        1118
24       22 bottom           1020.         260.
25       24 surface         13034.        1149
26       24 bottom            700.         195.
27       26 surface          9348.        1019
28       26 bottom            192.          80.7
29       28 surface          3578.         699
30       28 bottom             35.3         21.8
31       30 surface           669.         231
32       30 bottom              2.67         2.67
33       32 surface           144           30
34       34 surface            18            8
```

**Reply by jordansread** [reply]
Thanks Hayley. I continue to be most worried about the consistent summer bias _across_ GCMs that is larger than NLDAS for all GCMs. This worries me because the GCMs generally blanket reality (some are cold, some hot, others ok) and to see that all six are +2°C in the summer is worrisome. 

🤔

**Comment by hcorson-dosch-usgs** [comment]
Okay just completed a second complete re-run of the GCM models (to confirm the results of the 8/25 run) following a few changes to the `lake-temperature-model-prep` pipeline to 1) only pull GCM data for lakes that [fall within the GCM footprint](https://github.com/USGS-R/lake-temperature-model-prep/pull/350)  2) apply the [wind transform to the GCM drivers](https://github.com/USGS-R/lake-temperature-model-prep/pull/354), and 3) Update the `nml_list.rds` to reflect that two WI lakes now only have max depth instead of [faulty hpso](https://github.com/USGS-R/lake-temperature-model-prep/pull/358). There was also one change to `lake-temperature-process-models` to [fix a bug](https://github.com/USGS-R/lake-temperature-process-models/pull/77) where simulation directories for failed model runs were not deleted prior to new runs, and therefore meteo data for previously failed runs was not updated.

There are 12,299 lakes that fall within the CASC footprint and within the GCM footprint.

That amounts to 221382 unique model runs. 218970 (98.9%) succeeded. 

All 18 model runs succeeded for 11,715 lakes, or 95.3% of lakes. 1+ runs failed for 584 lakes.

The 2,412 model failures for 584 lakes were primarily due to NA values and model layers dropping to 1

```r
 p2_gcm_glm_uncalibrated_runs %>% left_join(p1_lake_cell_tile_xwalk_df, by='site_id') %>% filter(glm_success==FALSE) %>% group_by(glm_code, extraction_error) %>% summarize(n=n())
`summarise()` has grouped output by 'glm_code'. You can override using the `.groups`
argument.
# A tibble: 5 × 3
# Groups:   glm_code [3]
  glm_code extraction_error                                                    n
     <int> <chr>                                                           <int>
1        0 NA temperature values                                            1322
2        0 Number of layers drops to 1                                      1019
3        0 Number of layers drops to 1, Maximum hice value exceeds lake d…     6
4        1 NA                                                                 45
5      134 NA                                                                 20
```
Note that the 'NA temperature values' error is triggered by [the extraction of GLM temperatures](https://github.com/USGS-R/lake-temperature-process-models/blob/main/2_run/src/run_glm3.R#L17) and precedes [the generation of the other two extraction errors](https://github.com/USGS-R/lake-temperature-process-models/blob/main/2_run/src/run_glm3.R#L33-L48). So lakes for which temperature data extraction failed (`NA temperature values`) may have also have excessive ice generation or only 1 simulation layer.

**Reply by hcorson-dosch-usgs** [reply]
The mean lake depth across the 584 lakes for which 1+ GCM runs failed is 1.6 meters. Median lake depth is 1 m, minimum lake depth is 0 m (4 sites - c("nhdhr_113236931", "nhdhr_151870592", "nhdhr_151874009", "nhdhr_45361479" )), maximum lake depth is 23.8 m (`nhdhr_47712182`). In all, lake models failed for 18 sites deeper than 5 m.

**Reply by hcorson-dosch-usgs** [reply]
The updated GCM evaluation plots are [HERE](https://doimspp.sharepoint.com/:f:/r/sites/IIDDStaff/Shared%20Documents/Project%20-%20Modeling%20Lakes/GLM/lake-temperature-process-models/GCM_evaluation_plots/FullFootprint_20220901?csf=1&web=1&e=zRvRpm). There were 1,126 evaluation sites (sites with >= 10 min_obs_dates, successful model runs, and matched pred-obs) in the GCM contemporary period (1981-2000).

**Reply by hcorson-dosch-usgs** [reply]
The updated run summary, which includes an overall summary of NLDAS and GCM runs is [HERE](https://doimspp.sharepoint.com/:u:/r/sites/IIDDStaff/Shared%20Documents/Project%20-%20Modeling%20Lakes/GLM/lake-temperature-process-models/GLM_run_summary_20220901.html?csf=1&web=1&e=rbAZ5R)

**Reply by lindsayplatt** [reply]
Just want to share the DOY bias plots before and after the wind transform was applied:

![image](https://user-images.githubusercontent.com/13220910/188228726-ebd5050e-e86c-4434-8cf5-606356cd721e.png)

**Reply by lindsayplatt** [reply]
The bias on the bottom depths is now more warm not cool. I think we talked about this before and it makes sense for the wind speed changes because there is more mixing with the higher windspeeds, so the bottom depths temps would get warmer.

**Reply by hcorson-dosch-usgs** [reply]
Failed sites seem to be distributed across the footprint:
![image](https://user-images.githubusercontent.com/54007288/189928507-d272f8f0-b3e4-4d7b-a3a7-ab27de64fcd1.png)
